mlflow/server/js/src/experiment-tracking/pages/prompts/components/ChatMessageCreator.tsx

```diff
@@ -0,0 +1,95 @@
+import {
+  Button,
+  PlusIcon,
+  TrashIcon,
+  RHFControlledComponents,
+  useDesignSystemTheme,
+  FormUI,
+} from '@databricks/design-system';
+import { Fragment } from 'react';
+import { useFieldArray, useFormContext } from 'react-hook-form';
+import { FormattedMessage, useIntl } from 'react-intl';
+import type { ChatPromptMessage } from '../types';
+
+const SUGGESTED_ROLES = ['system', 'user', 'assistant'];
+
+/**
+ * Provides a small UI for composing chat-style prompts as a list of role/content pairs.
+ */
+export const ChatMessageCreator = ({ name }: { name: string }) => {
+  const { control, formState } = useFormContext<{ [key: string]: ChatPromptMessage[] }>();
+  const { fields, insert, remove, replace } = useFieldArray({ control, name });
+  const { theme } = useDesignSystemTheme();
+  const { formatMessage } = useIntl();
+
+  const addMessageAfter = (index: number) => insert(index + 1, { role: 'user', content: '' });
+
+  const clearAll = () => replace([{ role: 'user', content: '' }]);
+
+  return (
+    <div css={{ display: 'flex', flexDirection: 'column', gap: theme.spacing.sm }}>
+      {fields.map((field, index) => (
+        <div
+          key={field.id}
+          css={{ display: 'grid', gridTemplateColumns: '120px 1fr auto auto', gap: theme.spacing.sm }}
+        >
+          <Fragment>
+            <RHFControlledComponents.Input
```

Comment:
should we use TypeaheadCombobox for this?

```suggestion
            <RHFControlledComponents.TypeaheadCombobox
```